### PR TITLE
Add support for Select Menus in Modals

### PR DESCRIPTION
**Pull request by NeloBlivion**
## Summary

"Implements" select menus in modals (it was basically done already, just had to adjust `refresh_state` and type checking)
This isn't the neatest PR in terms of implementation; it does the job, but as Discord potentially adds more features to modals it may be ideal to have a new class to manage further modal children types.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by Lulalaby** [comment]
On hold till modal selects are released.

Selects for modals are only unofficially implemented in discord rn, it crashes old apps and on phone don't even render.

File: discord/ui/item.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def to_component_dict(self) -> Dict[str, Any]:
     def refresh_component(self, component: Component) -> None:
         return None
 
-    def refresh_state(self, interaction: Interaction) -> None:
+    def refresh_state(self, data) -> None:
```
Comment (EmreTech) [inline discord/ui/item.py:86]: I don't understand why this change was made

Comment (Lulalaby) [reply discord/ui/item.py:86]: yeah, it'll be always an interaction

Comment (NeloBlivion) [reply discord/ui/item.py:86]: This was part of adjusting the `refresh_state` methods across items to take a dict instead of `Interaction`, as this is how it was handled in InputText and the modal `dispatch` method; I figured this made more sense than changing modal dispatch to use an `Interaction` for `refresh_state`.
image

Comment (EmreTech) [reply discord/ui/item.py:86]: Alright

**Review by EmreTech — APPROVED** [review]

File: discord/ui/modal.py [hunk 2]
```diff
@@ -123,30 +124,30 @@ def key(item: InputText) -> int:
...
             The item to add to the modal dialog
         """
 
         if len(self._children) > 5:
             raise ValueError("You can only have up to 5 items in a modal dialog.")
 
-        if not isinstance(item, InputText):
-            raise TypeError(f"expected InputText not {item.__class__!r}")
+        if not (isinstance(item, InputText) or isinstance(item, Select)):
+            raise TypeError(f"expected [InputText, Select] not {item.__class__!r}")
```
Comment (plun1331) [inline discord/ui/modal.py:140]: Can be simplified
```suggestion
        if not isinstance(item, (InputText, Select)):
            raise TypeError(f"expected InputText or Select not {item.__class__!r}")
```

File: discord/ui/modal.py [hunk 3]
```diff
@@ -65,15 +66,15 @@ def title(self, value: str):
...
         return self._children
 
     @children.setter
-    def children(self, value: List[InputText]):
+    def children(self, value: List[Union[InputText, Select]]):
         for item in value:
-            if not isinstance(item, InputText):
-                raise TypeError(f"all Modal children must be InputText, not {item.__class__.__name__}")
+            if not (isinstance(item, InputText) or isinstance(item, Select)):
+                raise TypeError(f"all Modal children must be InputText or Select, not {item.__class__.__name__}")
```
Comment (plun1331) [inline discord/ui/modal.py:77]: Can be simplified

```suggestion
            if not isinstance(item, (InputText, Select)):
                raise TypeError(f"all Modal children must be InputText or Select, not {item.__class__.__name__}")
```

**Comment by NeloBlivion** [comment]
Small update on this, [a Discord admin claimed this was a bug](https://discord.com/channels/613425648685547541/697489244649816084/1009552426879164506) and hence this was disabled on the API side. We'll probably see modal selects appear officially in the future, but for now this PR will no longer work (merge conflicts aside).

**Comment by Nzii3** [comment]
> On hold till modal selects are released.
@Lulalaby Select Menus in modals have been implemented publicly

**Comment by plun1331** [comment]
> > On hold till modal selects are released.
> > @Lulalaby Select Menus in modals have been implemented publicly

Are you confusing this with the new select menus? To my knowledge they haven't released selects in modals yet.

**Comment by Lulalaby** [comment]
It isn't available for modals. Pr still on hold!

**Comment by Nzii3** [comment]
> > > On hold till modal selects are released.
> > > @Lulalaby Select Menus in modals have been implemented publicly
> 
> Are you confusing this with the new select menus? To my knowledge they haven't released selects in modals yet.

Yeah my bad I realized they are different things
